test/memkind_memtier_data_movement_test.cpp

```diff
@@ -59,3 +60,34 @@ TEST_F(MemkindMemtierDataMovementTest,
     m_tier_memory = memtier_builder_construct_memtier_memory(m_builder);
     ASSERT_EQ(nullptr, m_tier_memory);
 }
+
+TEST(MemkindMemtierDataMovementBgThreadTest, test_bg_thread_lifecycle)
```

Comment:
I thought about creating MTTAllocator only for this test... Please check other comments below